### ISSUE TITLE
multuGPU test results.

### ISSUE THREAD

**Issue by ptits** [jy0205/Pyramid-Flow#95]
![shot_241014_020315](https://github.com/user-attachments/assets/a6212ef9-85d2-4201-8a9a-e31d443022b1)

I have 8xA100 to test.

Here are some multuGPU results.

As I see it works best after duration > 16.

I use cli:
./scripts/app_multigpu_engine.sh 4 diffusion_transformer_768p ./pyramid_flow_model t2v 31 9 5 768p 31_4_output.mp4 prompt

**Comment by FurkanGozukara** [comment]
wow so it took 2 minute 11 seconds for 16 second video?

**Comment by ptits** [comment]
https://github.com/user-attachments/assets/a2f77f5f-0284-4c55-8588-55e6719774c5


https://github.com/user-attachments/assets/8e00097b-3131-4ac3-a9be-d908fe1d608a


https://github.com/user-attachments/assets/f33caf10-bc10-4985-b565-44d9eb8ca9e7


https://github.com/user-attachments/assets/b204f3c2-c05b-4a8c-aa22-db8bb9cf1808



> wow so it took 2 minute 11 seconds for 16 second video?

no, duration (for some reasons) means different values:
duration 16 is 5 seconds
duration 31 is 10 seonds

**Comment by ptits** [comment]
https://github.com/user-attachments/assets/886742a0-58a2-427e-8129-f0d554d2e0bb

**Comment by feifeiobama** [comment]
> wow so it took 2 minute 11 seconds for 16 second video?

No, I think it means `temp=16`, which corresponds to a 5 second video.

**Comment by ptits** [comment]
> > wow so it took 2 minute 11 seconds for 16 second video?
> 
> No, I think it means `temp=16`, which corresponds to a 5 second video.

yes, but in app.py there is "Duration" slider (which is actually temp)

this might be kinda confusing for users

![shot_241014_120205](https://github.com/user-attachments/assets/6c25a091-1363-4507-b08b-db235203b8ff)

**Comment by ptits** [comment]
And looks like 8 GPUs dont give the same boost as 4 does.

**Comment by aniolekx** [comment]
can you share code, how to split it across GPUs?

**Comment by ptits** [comment]
https://github.com/jy0205/Pyramid-Flow/blob/main/app_multigpu.py
